src/tiled/commanddialog.cpp

```diff
@@ -118,3 +135,8 @@ void CommandTreeView::removeSelectedCommands()
     const QModelIndexList indices = selection->selectedRows();
     mModel->removeRows(indices);
 }
+
+void CommandTreeView::currentChanged(const QModelIndex &current, const QModelIndex&)
```

Comment:
Rather than having this in the view, please handle this in the `CommandDialog`. This way you can also remove `mDialog`. To do this you should connect to the `currentChanged` signal of the `treeView->selectionModel()`.